contracts/Curation.sol

```diff
@@ -0,0 +1,415 @@
+            // Redistribute tokens
+            staking.unlockAndMoveTokens(challenge.challenger, challenge.lockId, application.applicant, reward);
+        } else { // challenge accepted (application rejected)
+            // it has been already registered
+            if (application.registered) {
+                // remove from Registry app
+                registry.remove(entryId);
+                application.registered = false;
+            }
+            // Remove challenger (winner) used lock
+            delete(usedLocks[challenge.challenger][challenge.lockId]);
+            // Unlock applicant tokens from Staking app
+            reward = application.amount.mul(dispensationPct) / PCT_BASE;
+            // Redistribute tokens
+            staking.unlockAndMoveTokens(application.applicant, application.lockId, challenge.challenger, reward);
+        }
+
+        ResolvedChallenge(entryId, voteResult);
+    }
+
+    function claimReward(bytes32 entryId) isInitialized public {
+        require(isChallengeResolved(entryId));
+
+        Challenge storage challenge = challenges[entryId];
+        Application memory application = applications[entryId];
+
+        // avoid claiming twice
+        require(!challenge.claims[msg.sender]);
+
+        // register claim to avoid claiming it again
+        challenge.claims[msg.sender] = true;
+
+        bool voteResult;
+        uint256 totalWinningStake;
+        (voteResult, totalWinningStake,) = voting.getVoteResult(challenge.voteId);
+
+        address loser;
+        uint256 loserLockId;
+        uint256 amount;
+        if (voteResult == false) {
+            loser = challenge.challenger;
+            loserLockId = challenge.lockId;
+            amount = challenge.amount;
+        } else { // voteResult == true
+            loser = application.applicant;
+            loserLockId = application.lockId;
+            amount = application.amount;
+        }
+
+        // reward as a voter
+        uint256 voterWinningStake = voting.getVoterWinningStake(challenge.voteId, msg.sender);
+        require(voterWinningStake > 0);
+        // amount * (voter / total) * (1 - dispensationPct)
+        uint256 reward = amount.mul(voterWinningStake).mul(PCT_BASE.sub(dispensationPct)) / (totalWinningStake * PCT_BASE);
+        // Redistribute tokens
+        staking.unlockAndMoveTokens(loser, loserLockId, msg.sender, reward);
+
+        // check if lock can be released
+        uint256 remainingLockAmount;
+        (remainingLockAmount, ) = staking.getLock(loser, loserLockId);
+        if (remainingLockAmount == 0) { // TODO: with truncating, this may never happen!!
+            delete(usedLocks[loser][loserLockId]);
+            // Remove application, if it lost, as redistribution is done
+            if (voteResult == true) {
+                delete(applications[entryId]);
+            }
+        }
+    }
+
+    function registerApplication(bytes32 entryId) isInitialized public {
```

Comment:
If this function can only be used for unchallenged applications, I would rename to signal that: `registerUnchallengedApplication(bytes32 entryId)`